src/main/java/org/example/service/TableService.java

```diff
@@ -0,0 +1,33 @@
+package org.example.service;
+
+import org.example.model.Developer;
+
+public class TableService {
+
+    private static final DeveloperService developers = DeveloperService.getInstance();
+
+    private static TableService INSTANCE;
+
+    public String getHtmlTable() {
+
+        String messageStart = "<html><head></head><center><body><table><th>user_id</th><th>firstName</th><th>lastName</th><th>age</th><th>progLang</th><tr>";
+
+        String finalMessage = "";
+
+        for (Developer developer : developers.getDevelopers()) {
+            String messagePattern = "<tr><td>%s</td><td>%s</td><td>%s</td><td>%s</td><td>%s</td></tr>";
+            finalMessage += String.format(messagePattern, developer.getUserId(), developer.getFirstName(), developer.getSecondName(), developer.getAge(), developer.getProgLang());
```

Comment:
Do not use concatenation.
Check `StringBuilder` class from the JavaCore.

Reply:
fixed